languages/rust/exercises/concept/vec-macro/tests/vec_macro.rs

```diff
@@ -0,0 +1,21 @@
+#[cfg(test)]
+mod tests {
+    use vec_macro::*;
+    #[test]
+    fn test_empty() {
+        assert_eq!(create_empty(), Vec::new());
+    }
+    #[test]
+    fn test_buffer() {
+        let mut zeroized = Vec::new();
+        zeroized.extend_from_slice(&[0, 0, 0]);
+        assert_eq!(create_buffer(3), zeroized);
+    }
```

Comment:
```suggestion
    fn test_buffer() {
        for n in 0..10 {
            let zeroized = create_buffer(n);
            assert_eq!(zeroized.len(), n);
            assert!(zeroized.iter().all(|v| v == 0));
        }
    }
```

Test the properties we care about instead of creating a replica to assert equality.